### PR TITLE
add a new field `middle_key` for region properties.

### PR DESCRIPTION
**Pull request by hicqu**
Add a new field `middle_key` in region properties.

### CODE REVIEW

**Comment by huachaohuang** [comment]
Why do you need to add this?

**Comment by hicqu** [comment]
Sometimes we want to use `split_region` API. So we need to get approximate middle key by this RPC.

**Comment by huachaohuang** [comment]
I remember there is a half split API?

**Comment by hicqu** [comment]
TiKV only have a `split_region` API, which should be used with a given split_key. So how to calculate the key can be controled by caller. Maybe we can re-implement `HalfSplit` by the approximate key instead of scan region later, however let `get_region_properties` can also return the middle key is helpful for now.

**Comment by hicqu** [comment]
ping @huachaohuang  @zhangjinpeng1987

File: proto/debugpb.proto [hunk 1]
```diff
@@ -214,4 +214,6 @@ message GetRegionPropertiesRequest {
 
 message GetRegionPropertiesResponse {
     repeated Property props = 1;
-}
\ No newline at end of file
+    // The middle key of the region in encoded format, without ts padding.
+    bytes middle_key = 2;
```
Comment (zhangjinpeng1987) [inline proto/debugpb.proto]: You can use 'middle_key_by_approximate_size' as an `props`'s name.

Comment (hicqu) [reply proto/debugpb.proto]: Yes, so the PR can be closed now.

**Pull request closed without merging**